docs/okeps/okep-5259-no-overlay.md

```diff
@@ -0,0 +1,1459 @@
+    apiVersion: k8s.ovn.org/v1
+    kind: RouteAdvertisements
+    metadata:
+      name: blue
+    spec:
+      # nodeSelector must be empty, since we don't support a network in an overlay and no-overlay hybrid mode.
+      nodeSelector: {}
+      frrConfigurationSelector:
+        matchLabels:
+        # Select the FRRConfiguration defined in step-3
+        network: blue
+      networkSelectors:
+      - networkSelectionType: ClusterUserDefinedNetwork
+        clusterUserDefinedNetworkSelector:
+          networkSelector:
+            matchLabels:
+              # Select the CUDN defined in step-2
+              network: blue
+      advertisements:
+      - PodNetwork
+    ```
+
+5. The UDN controller in ovnkube-controller will create the OVN virtual topology
+   for the UDN. Neither the transit switch nor the static routes to remote pod
+   subnets will be created.
+
+6. The UDN controller in ovnkube-cluster-manager will watch the
+   RouteAdvertisements events to check whether at least one RouteAdvertisements
+   CR is correctly configured to advertise the pod networks and its status is
+   `Accepted`. The UDN controller will update the status condition
+   `TransportAccepted` accordingly.
+
+7. The BGP feature is responsible for exchanging BGP routes to remote pod
+   subnets between nodes. Ultimately, these routes will be imported into the
+   gateway router.
+
+#### Create a ClusterUserDefinedNetwork in No-overlay Mode with Managed Routing
+
+1. The frr-k8s pods shall be deployed on each node.
+
+2. In the Managed routing mode, a new FRRConfiguration controller will be
+   responsible for creating the base FRRConfiguration CR which sets up the
+   cluster internal BGP fabric. The cluster admin only needs to start
+   ovn-kubernetes with the following flags:
+
+    ```ini
+    [bgp]
+    topology = full-mesh
+    as-number = 64514
+    ```
+
+   The generated FRRConfiguration will be the same as the one described in the
+   managed default network section.
+
+3. A cluster admin wants to enable no-overlay mode for the blue network by
+   creating the following ClusterUserDefinedNetwork CR.
+
+    ```yaml
+    apiVersion: k8s.ovn.org/v1
+    kind: ClusterUserDefinedNetwork
+    metadata:
+      name: blue
+      labels:
+        network: blue
+    spec:
+      namespaceSelector:
+        matchExpressions:
+        - key: kubernetes.io/metadata.name
+          operator: In
+          values: ["ns1", "ns2"]
+      network:
+        topology: Layer3
+        layer3:
+          role: Primary
+          # The UDN MTU shall not be larger than the provider network's MTU.
+          mtu: 1500
+          subnets:
+          - cidr: 10.10.0.0/16
+            hostSubnet: 24
+        transport: "NoOverlay"
+        noOverlayOptions:
+          outboundSNAT: "Enable"
+          routing: "Managed"
+    ```
+
+4. The UDN controller in ovnkube-cluster-manager will create a
+   RouteAdvertisements CR named `blue` that advertises the blue network.
```

Comment:
Again, maybe important to avoid generating common RouteAdvertisements CR names to avoid the potential for conflict.